TestFoundation/TestThread.swift

```diff
@@ -84,4 +88,44 @@ class TestThread : XCTestCase {
         XCTAssertEqual(thread3.name, "Thread3")
         XCTAssertNotEqual(thread3.name, getPThreadName())
     }
+
+    func test_mainThread() {
+        XCTAssertTrue(Thread.isMainThread)
+        let t = Thread.mainThread
+        XCTAssertTrue(t.isMainThread)
+        let c = Thread.current
+        XCTAssertTrue(c.isMainThread)
+        XCTAssertTrue(c.isExecuting)
+        XCTAssertTrue(c.isEqual(t))
+
+        var started = false
+        let condition = NSCondition()
+        let thread = Thread() {
+            condition.lock()
+            started = true
+            XCTAssertFalse(Thread.isMainThread)
+            XCTAssertFalse(Thread.mainThread == Thread.current)
+            condition.broadcast()
+            condition.unlock()
+        }
+        thread.start()
+
+        condition.lock()
+        if !started {
```

Comment:
I removed the `started` flag and used `wait(until:)` (including fixing a bug with the implementation) and then tried some tests including not starting the thread or making the thread take too long, and the timeout worked fine under those conditions.

I gave it 10seconds timeout which should cover all cases except incredibly high server load and if thats the case -CI will have other problems anyway